core/version/gear.ex

```diff
@@ -32,12 +32,18 @@ defmodule AntikytheraCore.Version.Gear do
           # Croma defines modules (e.g. `Elixir.Croma.TypeGen.Nilable.Antikythera.Email`) automatically in the gear.
           # As a result, some modules which have the same name are defined in two or more gears.
           # We have to avoid loading these modules twice.
-          skip_to_load? = can_skip_to_load?(mod) && :code.is_loaded(mod)
-          if not skip_to_load? do
-            case :code.load_file(mod) do
-              {:module, _}      -> :ok
-              {:error , reason} -> raise "Failed to load '#{mod}': #{reason}"
+          if auto_generated_module?(mod) do
+            if !:erlang.module_loaded(mod) do
+              case :code.load_file(mod) do
+                {:module, _}           -> :ok
+                # When three or more same-name modules are loaded at the same time,
+                # `:code.load_file(mod)` returns `:not_purged`. We can safely ignore the error in that case.
+                {:error , :not_purged} -> :ok
+                {:error , reason}      -> raise "Failed to load '#{mod}': #{reason}"
+              end
             end
+          else
+            {:module, _} = :code.load_file(mod)
           end
```

Comment:
(optional)
I think writing shorter case first is more readable.

```elixir
          if not auto_generated_module?(mod) do
            {:module, _} = :code.load_file(mod)
          else
            if !:erlang.module_loaded(mod) do
              case :code.load_file(mod) do
                {:module, _}           -> :ok
                # When three or more same-name modules are loaded at the same time,
                # `:code.load_file(mod)` returns `:not_purged`. We can safely ignore the error in that case.
                {:error , :not_purged} -> :ok
                {:error , reason}      -> raise "Failed to load '#{mod}': #{reason}"
              end
            end
          end
```

Reply:
Fixed be13b4df2c711637193077408f3060fea4c9efc4